src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/RemoveResourceDesignerStep.cs

```diff
@@ -0,0 +1,211 @@
+using Mono.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+using System;
+using System.Linq;
+using Xamarin.Android.Tasks;
+using System.Collections.Generic;
+using Mono.Cecil.Cil;
+using System.Text.RegularExpressions;
+
+namespace MonoDroid.Tuner
+{
+	public class RemoveResourceDesignerStep : BaseStep
```

Comment:
An idea that could be done in a future iteration.

Could this step also modify this class:

https://github.com/xamarin/xamarin-android/blob/7ba62bac5a25eb783a6da9c2b6a989a820695a1f/src/Mono.Android/Android.Runtime/ResourceIdManager.cs#L6-L9

If `UpdateIdValues()` was a completely empty method, some reflection could be skipped at startup.

Reply:
Depending on when the new Step runs, it might strip out that class already since the Designer is the thing that calls`UpdateIdValues`. But its worth doing in the future yes :)

Reply:
@dellis1972 [wrote](https://github.com/xamarin/xamarin-android/pull/5317#discussion_r575295508):

> the Designer is the thing that calls `UpdateIdValues`.

This is incorrect.  All (current) `Resource.designer.cs` code calls `UpdateIdValues`:

https://github.com/xamarin/xamarin-android/blob/214dd76d9441c9974775f0b29feec0d5d8d1097f/src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceDesigner.cs#L185-L192

https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Expected/GenerateDesignerFileExpected.cs

(which contains *21* references to `UpdateIdValues()`.

Any use of a `Resource.Descriptor.Name` field will cause `UpdateIdValues()` to be invoked.

Now, if after this "link resource nested types" change operates, the static constructor is *also* removed, meaning that `UpdateIdValues()` is likewise not invoked anymore, that would be cool.

If that's the case, though, that should *also* be mentioned in the PR/commit message.  That's a possibly important (and huge) semantic change.  (Not necessarily a "break", but certainly something we'll need to know about.)